### ISSUE TITLE
Matchmaker in Nakama 3.1.2 won't find any matches (whereas 2.15.0 does)

### ISSUE THREAD

**Issue by dsnopek** [heroiclabs/nakama#586]
## Description
I have an existing game that works great with Nakama 2.15.0 (and also worked fine with previous versions of Nakama 2.x).

I tried upgrading my local development Nakama to 3.1.2, and everything seemed to just work, except for matchmaking. For the most part, it simply fails to find a match (whereas with 2.15.0 it matches immediately). However, I did have it successfully match exactly one time. I don't know what was different about that one time, I was just trying over and over, and it was sandwiched in between a dozen failures.

## Steps to Reproduce

Here's a simplified version of the GDScript code that's running on both clients:

```
nakama_socket.add_matchmaker_async("+properties.game:retro_tank_party1 +properties.client_version:dev", 2, 4, {client_version = "dev", game = "retro_tank_party1"})
```

It does give a successful result with a matchmaker ticket - it just never finds a match.

## Expected Result

A match is found.

## Actual Result

No match is found, even though I have two clients connected that should find each other.

## Context
- [ ] Unity
- [ ] Unreal
- [X] Other (Godot)

## Your Environment

- Nakama: 3.1.2
- Database: 19.2.5
- Environment name and version: Docker 19.03.13
- Operating System and version: Ubuntu 18.04

**Comment by novabyte** [comment]
@dsnopek In Nakama 3.0 the server's matchmaker implementation was updated to include support for party-based matchmaking. It also introduces new configuration options to control how fast the matchmaker converges to match users:

```
  -matchmaker.interval_sec int
    	How quickly the matchmaker attempts to form matches, in seconds. (default 15)
  -matchmaker.max_intervals int
    	How many intervals the matchmaker attempts to find matches at the max player count, before allowing min count. (default 2)
  -matchmaker.max_tickets int
    	Maximum number of concurrent matchmaking tickets allowed per session or party. (default 3)
```

What do these settings look like for your game?

**Comment by dsnopek** [comment]
Aha!

It's using the default values for all those settings. I tried again, and if I waited for ~30 seconds, it would finally match the two clients. I tried changing `matchmaker.interval_sec` to 2, and it matched in ~4 seconds. I'm just used to it matching instantaneously in testing, and apparently, I never waited as long as 30 seconds. So, this was a bug in my expectations, not Nakama. :-)

Is there a "Upgrading to Nakama 3" document somewhere that describes changes to look out for when upgrading?

In any case, this issue can be closed.
